boefjes/boefjes/api.py

```diff
@@ -0,0 +1,177 @@
+import base64
+import logging
+import multiprocessing
+from datetime import datetime, timezone
+from enum import Enum
+from typing import List, Optional
+
+from fastapi import FastAPI, HTTPException, Response
+from pydantic import BaseModel, Extra, Field
+from requests import HTTPError
+from uvicorn import Config, Server
+
+from boefjes.clients.bytes_client import BytesAPIClient
+from boefjes.clients.scheduler_client import SchedulerAPIClient, TaskStatus
+from boefjes.config import settings
+from boefjes.job_handler import (
+    _collect_default_mime_types,
+    _find_ooi_in_past,
+    get_environment_settings,
+    get_octopoes_api_connector,
+    serialize_ooi,
+)
+from boefjes.job_models import BoefjeMeta
+from boefjes.katalogus.local_repository import get_local_repository
+from octopoes.models import Reference
+
+app = FastAPI()
+logger = logging.getLogger(__name__)
+local_repository = get_local_repository()
+scheduler_client = SchedulerAPIClient(settings.scheduler_api)
+bytes_api_client = BytesAPIClient(
+    settings.bytes_api,
+    username=settings.bytes_username,
+    password=settings.bytes_password,
+)
+
+
+class UvicornServer(multiprocessing.Process):
+    def __init__(self, config: Config):
+        super().__init__()
+        self.server = Server(config=config)
+        self.config = config
+
+    def stop(self):
+        self.terminate()
+
+    def run(self, *args, **kwargs):
+        self.server.run()
+
+
+def run():
+    config = Config(app, host=settings.boefje_api_host, port=settings.boefje_api_port)
+    instance = UvicornServer(config=config)
+    instance.start()
+    return instance
+
+
+class BoefjeInput(BaseModel):
+    task_id: str
+    output_url: str
+    boefje_meta: BoefjeMeta
+
+    class Config:
+        extra = Extra.forbid
+
+
+class StatusEnum(str, Enum):
+    COMPLETED = "COMPLETED"
+    FAILED = "FAILED"
+
+
+class File(BaseModel):
+    name: Optional[str]
+    content: str = Field(..., contentEncoding="base64")
+    tags: Optional[List[str]]
+
+    class Config:
+        schema_extra = {
+            "required": ["content", "tags"],
```

Comment:
Tags is defined as optional, but required here?

Reply:
Good point, it shouldn't be required as we have default tags (MIME types) for each boefje. I removed the `schema_extra.required` entirely because the typing (`Optional` or not) already covers what fields are required.